web/src/pages/your_account/mod.rs

```diff
@@ -70,29 +66,23 @@ impl Component for YourAccount {
             }
             Self::Message::ClickUser(ref username) => {
                 let user = self.users.get(username).unwrap().to_owned();
-                self.send_transaction_builder.set_user(user);
-                self.send_transaction_dialog.show();
-                false
-            }
-            Self::Message::UpdateAmountToSend(amount) => {
-                self.send_transaction_builder.set_amount(amount);
+
+                self.dialog_user = Some(user);
+                self.dialog_link.show();
                 false
             }
-            Self::Message::SendTransaction => {
-                let builder = mem::take(&mut self.send_transaction_builder);
-                let user = self.props.user.clone();
-                let (recipient, amount) = builder.build().unwrap();
+            Self::Message::SendTransaction((recipient, amount)) => {
                 let sequence = 1; // TODO retrieve latest sequence
 
-                self.send_asset_agent
-                    .send((user, sequence, recipient, amount as u64));
+                self.send_asset_agent.send((
+                    self.props.user.clone(),
+                    sequence,
+                    recipient,
+                    amount as u64,
```

Comment:
Is that a safe assumption here? Or is `amount` user-input and might be tainted?

Reply:
It is for the target we are using: `amount` is a `usize`, which is "on a 32 bit target, this is 4 bytes and on a 64 bit target, this is 8 bytes.".

https://doc.rust-lang.org/std/primitive.usize.html

Adding a TODO for now to handle it when fixing all the TODOs.